### DISCUSSION TITLE
Release 1.2.0 Beta 1

### DISCUSSION THREAD

**Discussion by apohl79** [apohl79/audiogridder#495] [General]
## GENERAL
- **NEW** Sandboxing: Each AG plugin (chain) will run in a dedicated server process.
  - A  crashed remote plugin will only affect a single AG chain, rather than killing the entire server.
  - Complex sessions load much faster due to real parallel loading of remote plugins.
  - The feature has to be enabled in the server settings.
- **NEW** Local mode: Improved UI experience when running server and DAW on the same computer. (Use this in combination with sandboxing to have multiple parallel plugin UIs!)
- **NEW** Sidechain and multi-channel (16in/16out) support.
- **Improvements:**
  - Plugin/Server handshake: True passive mode, all connections will be opened from the plugin to the server, which will allow a server to be behind a gateway and will simplify the network setup.
  - Added universal binaries for macOS.

## PLUGIN
- **NEW** Presets:
  - Store/Load FX chains or instruments as presets.
  - Set an FX preset as default to be auto-loaded with a new AG plugin instance.
- **NEW** Tray App:
  - Consolidating FX and Instrument plugins into one monitor window.
  - Support for reconnecting all plugins in one step.
- **NEW** Automation assign/unassign all  command.
- **Improvements:**
  - Always open a Plugin UI.
  - Use Generic Editor automatically when a plugin has no UI.
  - Improve UI contrast for activated insert.
  - Show the server address (IP) for multiple servers with same name.

:exclamation: **INSTALLATION NOTE:** On Windows you have to uninstall an existing AudioGridder Plugin before installing 1.2 :exclamation: 

## SERVER
- Added close button for plugin windows.
- Fix: Disable screen recording permission check for macOS 10.7

<hr /><em>This discussion was created from the release <a href='https://github.com/apohl79/audiogridder/releases/tag/release_1_2_beta1'>Release 1.2.0 Beta 1</a>.</em>

**Comment by emack999** [comment]
The plugin freezes the moment you add it in any DAW---Windows version, 20H2 19042.928 build.
Occurs in both Cubase 9.5 and Gig Performer 3.

You select Audiogridder in the DAW and then the DAW stops responding.  Can't even get to adding a plugin to AG.
Tried running server normally and also in Local mode---same result.
Confirmed that both server and plugin are version 1.2.0 beta1.
Occurs using both vst and vst3 versions of AG.

**Reply by apohl79** [reply]
Can you provide a bug report? Need to see logs etc.

**Reply by emack999** [reply]
From the text file, I see that it's hung up looking for the server at a particular IP, and not finding it.
[AudioGridderPlugin_2021-04-17_15-26-36.log](https://github.com/apohl79/audiogridder/files/6330077/AudioGridderPlugin_2021-04-17_15-26-36.log)

**Reply by apohl79** [reply]
Have you used the installer? Looks like the tray app was not installed under "C:\Program Files\AudioGridderPluginTray".

**Reply by emack999** [reply]
Uninstalling AG Server and Plugin versions and reinstalling fixed the issue.  It's possible that 1.1.1 needs to be uninstalled prior to 1.2 beta1 being installed for the installation to be successful.

**Reply by apohl79** [reply]
Ok I will try that, I hope this is not the case, at least it is not supposed to be.

**Reply by apohl79** [reply]
looks like you are right, for some reason, the installer does not install the tray app to its correct location when upgrading, only on a clean install.

**Comment by CaBuffalo** [comment]
Take a look at the firewall on your workstation. The ADApp tray needs to run.

**Reply by emack999** [reply]
No issues with firewall, settings exactly the same as they were for 1.1.1

**Comment by CaBuffalo** [comment]
No need to uninstall 1.1.1.

Anyways, it's good to hear that it's working for you now. Exploring and
enjoy.

On Sat, Apr 17, 2021 at 2:40 PM emack999 ***@***.***> wrote:

> Uninstalling AG and reinstalling fixed the issue. It's possible that 1.1.1
> needs to be uninstalled prior to 1.2 beta1 being installed for the
> installation to be successful.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/495#discussioncomment-625050>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6DUPUHJN5FQ5CJIAY3TJH54TANCNFSM43DN7NNA>
> .
>


-- 
Kevin Ma
Supported by Gmail.

**Reply by emack999** [reply]
@CaBuffalo As noted above by @apohl79 , you actually do have to uninstall 1.1.1 on Windows to get a proper installation of 1.2.0 beta1.

**Comment by emack999** [comment]
Ok---next issue is that AG Instruments produce no sound!  I don't think that MIDI signals are getting through from the client DAW to the plugin on the AG server.

Even manually manipulating keys on the plugin UI doesn't produce any sound.

**Reply by apohl79** [reply]
Do you see midi notes going through when you hit play in your DAW?

I have optimized the resource usage of the AG plugin by avoiding data transmission when the host is not playing/recording. I was probably a bit fast in considering the side effects.

**Reply by emack999** [reply]
Yes, when the playhead is engaged, the MIDI notes go through and notes manually played on the plugin UI make sound also.  So the issue is as you mentioned:  No data transmission when playhead is not engaged.

My use case is that I am using Gig Performer, which isn't a recording DAW, but more of a plugin and scripting host.  I play live, and as such I don't always engage the playhead on that app.  I'm sure there are many other use cases out there where the playhead would not be engaged, but the user would still want midi and sound to transmit to and from the AG plugin back to the client.  You should probably reconsider the decision to cut data transmission when the playhead is not engaged.

**Reply by apohl79** [reply]
I added an option for the new behaviour and reverted the default back to 1.1 (always transmitting data). Will be going into the next release.

**Reply by emack999** [reply]
@apohl79 Excellent, thank you for the response and help with finding the root cause of the issue.

**Reply by dhstsw** [reply]
Same here (Win10 server and client, ableton live 10).

**Comment by CaBuffalo** [comment]
I used the drum replacement on my snare. Then tried a random instrument. It
works fine.

Can you try different instruments and see how it goes?

On Sat, Apr 17, 2021 at 3:30 PM emack999 ***@***.***> wrote:

> Ok---next issue is that AG Instruments produce no sound! I don't think
> that MIDI signals are getting through from the client DAW to the plugin on
> the AG server.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/495#discussioncomment-625127>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6BDMXZXNDVJ6IZXQITTJIDWXANCNFSM43DN7NNA>
> .
>


-- 
Kevin Ma
Supported by Gmail.

**Reply by emack999** [reply]
Have tried 3 different instruments.  Like I said, even manual pressing of keys in a piano/organ/synth plugin UI doesn't make any sound.  Tried in Cubase 9.5, Reaper 6.27 and Gig Performer 3.

**Reply by emack999** [reply]
Here's a video showing the MIDI signal registering in Reaper, but nothing in Lounge Lizard midi monitor, and no sound generated by manually pressing the notes in the plugin UI.
https://imgur.com/a/afiPpCc

Here's a second video, this time using Gig Performer and Kontakt.  Same result.
https://imgur.com/a/EN5hgFj

**Comment by Fittkenny** [comment]
Brilliant update!!! :-) amazing work...

**Comment by CaBuffalo** [comment]
Yes.

On Sun, Apr 18, 2021 at 6:20 AM thelux68 ***@***.***> wrote:

> sorry but for Catalina and Big Sur I've to install x86 64?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/495#discussioncomment-626411>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6BSLIWUP7AVWVOCL4DTJLMCFANCNFSM43DN7NNA>
> .
>


-- 
Kevin Ma
Supported by Gmail.

**Comment by Fittkenny** [comment]
Hmm...Logic is really crashy with this new update. Just hardcrashes when I press the record arm button for exemple. Or when I try to redirect from local to external server the tray app goes berserk. Will investigate further. :-)

**Reply by apohl79** [reply]
Ok thx for trying it further. I'm testing with logic too. Please provide logs/traces if you can, and core files (if are not on big sur yet).

**Reply by Fittkenny** [reply]
I sent the log/trace files via wetransfer .....

**Reply by apohl79** [reply]
Do you see crash reports of logic under ~/Library/Logs/DiagnosticReports/ ?

(To navigate there, open finder, press cmd+shift+g and paste the location into the dialog)

**Reply by Fittkenny** [reply]
Oh, yes. Here are those with the correct time stamps I think:
[crashreports.zip](https://github.com/apohl79/audiogridder/files/6333832/crashreports.zip)

**Comment by fojerhar** [comment]
[AudioGridderPlugin_2021-04-19_16-10-31(2).log](https://github.com/apohl79/audiogridder/files/6336690/AudioGridderPlugin_2021-04-19_16-10-31.2.log)
[AudioGridderPlugin_2021-04-19_16-10-31.log](https://github.com/apohl79/audiogridder/files/6336691/AudioGridderPlugin_2021-04-19_16-10-31.log)
[AudioGridderPlugin_2021-04-19_16-07-39.log](https://github.com/apohl79/audiogridder/files/6336692/AudioGridderPlugin_2021-04-19_16-07-39.log)
[AudioGridderPlugin_2021-04-19_16-07-27.log](https://github.com/apohl79/audiogridder/files/6336693/AudioGridderPlugin_2021-04-19_16-07-27.log)
[AG 1.2.0 crash.zip](https://github.com/apohl79/audiogridder/files/6336699/AG.1.2.0.crash.zip)





HI
For me protools 2021.3.1 collapses immediately if I open the 1.2.0 plugin in a new session, (macos catalina 10.15.7), and if I open a session what was made with 1.1.1 it runs, as i stop playback pt freezes with beachball, but improvements seems nice.
Protools crash also attached

**Comment by CaBuffalo** [comment]
Hi, guys, do you see the AG App tray up and running?

Also, please check if you are running at the Sandbox mode checked. Give it a try if not.

**Reply by fojerhar** [reply]
Yes, tray app running fine, sandbox, local checked, the same here
[AudioGridderPlugin_2021-04-19_16-43-20.log](https://github.com/apohl79/audiogridder/files/6336934/AudioGridderPlugin_2021-04-19_16-43-20.log)
<img width="875" alt="Screenshot 2021-04-19 at 16 43 59" src="https://user-images.githubusercontent.com/72139943/115255613-bad4ae00-a12e-11eb-9fd8-b90d626b5825.png">

**Reply by apohl79** [reply]
Hey, if you would be able to collect a full bug report, that would be very helpful. I'm currently not able to reproduce this, but I was able to track down the logic crashes @Fittkenny reported. So I'm likely having a fix, but I can't verify it yet. So if you can, please follow https://audiogridder.com/bug-reports/ and provide logs/traces and crash reports from the ~/Library/Logs/DiagnosticReports
folder. Thx!

**Reply by dhstsw** [reply]
Yup, server icon is there (and server running "normally").
Tried both with and wihtout sandboxing, same result.
Hope it helps.
As for the logs, i'll do it asap (i'm not in the studio now).

**Reply by fojerhar** [reply]
Hi
I don't have any crashreports at L/L/Diagnosticreports, just the protools own crashreport what i have attached earlier.
And also could not create coredump with ptools.
Just found the plugin and server logfiles
[Crashreport.zip](https://github.com/apohl79/audiogridder/files/6338268/Crashreport.zip)

**Reply by apohl79** [reply]
Thx, looks like pro tools it catching crashes with a custom crash reporter. Thanks for checking. One final ask, can you enable tracing in the plugin as well? Thx!

**Reply by fojerhar** [reply]
Oh yes, I did not recognize it that the plugin also has a trace function.
Hope these datas can help you
Btw this is an amd hackintosh if it matters to you (i only had problems with reaper in this machine, but there is a patched version for amd hacks)
[Crashreport2.zip](https://github.com/apohl79/audiogridder/files/6338489/Crashreport2.zip)

**Reply by apohl79** [reply]
thx

**Comment by SymphonX** [comment]
Setup: [SERVER] Windows 7 (native and Wine) to [PLUGIN] MacOS 10.15.7, Logic Pro 10.6.1

With AG v1.2.0b1 I can access a max. of 16 automation parameters. Previous versions (particularly v1.1.b10) allowed for > 24 parameters. I need that amount of automation to control NeuronVS. Was the new limit implemented by design or an oversight? Otherwise everything appears to work well (even when running the server in Wine).

**Reply by apohl79** [reply]
No changes here, except for the assign all function. You can increase the number of automation slots manually in the plugin config file, see https://audiogridder.com/plugin/#automation

**Reply by SymphonX** [reply]
Ouch, I missed the cfg on the plug-in end (thought there was one only for the server). Thanks so much! This update renders AG perfect for my purposes!

**Comment by MatitaPere** [comment]
Hi,

Does AG store any plugin information when scanning? I might be crazy but I think my Hard Drive shrunk after scanning all plugins folders...
Another question, when I'm using the AG plugin I can only change the parameters in AG server. Is that how it's supposed to go or can I adjust inside the DAW?

Thank you for this great software!

**Reply by apohl79** [reply]
Regarding disk usage, please check if this thread #390 helps.

> Another question, when I'm using the AG plugin I can only change the parameters in AG server. Is that how it's supposed to go or can I adjust inside the DAW?

Yes you can control the plugin from your DAW remotely. For this the server needs the accessibility permission on macOS. Check that you have it. (see https://audiogridder.com/server/#install)

**Reply by MatitaPere** [reply]
Cool
